### ISSUE TITLE
external/XNNPACK: optimization

### ISSUE THREAD

**Issue by enh-google** [google/android-riscv64#42]
this whole project is mostly optimized assembler :-)

Zvediv looks like it would be very useful for them.

**Comment by rednoah91** [comment]
Looks like there are some RVV kernels are implemented: 
https://github.com/google/XNNPACK/commit/aac51aad45a24f1a1acec0a521ac63b6892028b6#diff-3f710269879983f53830fec90d0963ef9da0382cd4eed8bcaf699e8f4101560b

**Comment by rednoah91** [comment]
@fbarchard Are you working on it?

**Comment by fbarchard** [comment]
I am, but time frame is roughly by end of year.  I plan to focus on a full set of fp32 microkernels first.

**Comment by sharvey-img** [comment]
@fbarchard I see you've already done some work here (e.g. https://github.com/google/XNNPACK/commit/ccfee7043a30babb9e3aec70b342fcc743a260d2 ). Can we pick this up? Are there any non-merged commits?

**Comment by nick-knight** [comment]
We (SiFive) are also interested in contributing. My team is in the process of migrating (micro)kernels from our performance libraries into various open-source projects.

**Comment by JaydeepIMG** [comment]
Hi @nick-knight  @fbarchard, we at Imagination Technologies are also planning to contribute. Could you please share list of kernels that you are planning to implement? We are planning to start with 8bit quantized operations for RVV.

**Comment by ManojIMG** [comment]
Hi, We (Imagination) have created our first PR (QS8/QU8 RVV) here https://github.com/google/XNNPACK/pull/5638 but it seems waiting on the internal review. Is there anything missing from us to get it through? What is the expected turnaround time?
